dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -0,0 +1,224 @@
+package org.bitcoins.dlc.oracle
+
+import org.bitcoins.commons.jsonmodels.dlc.SigningVersion._
+import org.bitcoins.core.config.BitcoinNetwork
+import org.bitcoins.core.crypto.ExtKeyVersion.SegWitMainNetPriv
+import org.bitcoins.core.crypto.{ExtPrivateKeyHardened, MnemonicCode}
+import org.bitcoins.core.hd._
+import org.bitcoins.core.protocol.Bech32Address
+import org.bitcoins.core.protocol.script.P2WPKHWitnessSPKV0
+import org.bitcoins.core.util.TimeUtil
+import org.bitcoins.crypto._
+import org.bitcoins.dlc.oracle.storage._
+import org.bitcoins.keymanager.{DecryptedMnemonic, WalletStorage}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
+    val conf: DLCOracleAppConfig) {
+
+  implicit val ec: ExecutionContext = conf.ec
+
+  // what was previously used so we other wallets could find funds
+  //  private val signingKeyHDAddress = {
+//    val coin = HDCoin(HDPurposes.SegWit, HDCoinType.fromNetwork(conf.network))
+//    val account = HDAccount(coin, 0)
+//    val chain = HDChain(HDChainType.External, account)
+//    HDAddress(chain, 0)
+//  }
+
+  // 585 is a random one I picked, unclaimed in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+  private val PURPOSE = 585
+
+  private val rValueAccount: HDAccount = {
+    val coin = HDCoin(HDPurpose(PURPOSE), HDCoinType.fromNetwork(conf.network))
+    HDAccount(coin, 0)
+  }
+
+  private val chainIndex = 0
+
+  private def signingKey: ECPrivateKey = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+    // Use a different chain index so we don't reuse a key
+    val chain = chainIndex + 1
+
+    val path = BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chain'/0'")
```

Comment:
should call `getPath`

Reply:
If we use `getPath` it'll use the r value chain index instead of the signing key one.